### DISCUSSION TITLE
Stress/Strain measures

### DISCUSSION THREAD

**Discussion by jeyci94** [KratosMultiphysics/Kratos#12494] [Q&A]
What determines the stress/strain measures an element (or a simulation case) is working with?
If I wanted to change measures I am working with, how can this be done?

Opposite to the framework an element is working in (found as part of its _Parameters specifications_ in the implementation), it is not completely clear to me how to know when I am working with a specific measure.
Of course, if in a `Calculate` or `MaterialResponse` function, a function named `StrainGreenLagrange` is found it is clear, but there are parts of an element's implementation where e.g. a `CalculateOnIntegrationPoints` function takes an rVariable input where an internal `if` condition is for a "CAUCHY_STRESS_VECTOR" variable, it is not clear to me under which circumstances this is executed. Does this depend on the solicited output variables in the ProjectParameters.json?

How can I be sure of the stress and strain measures I am working with, and if I wanted to get results in a different one, e.g. from Green Lagrange to logarithmic Hencky strains, or use Kirchhoff stresses how could I achive this?
Would appreciate any suggestions or pointers, thanks!

**Comment by loumalouomega** [comment]
The strain/stress considered in each element depends of the kind of element (sic). No it is not the same an element that is total lagrangian or updated lagrangian for example. See our reference:

![61oqme06sIL _AC_UF1000,1000_QL80_](https://github.com/KratosMultiphysics/Kratos/assets/19991680/0feab19b-0516-418d-adef-7eb9a781a119)

"Nonlinear Finite Elements for Continua and Structures, Second Edition," authored by Ted Belytschko, Wing Kam Liu, Brian Moran, and Khalil I. Elkhodary, and published by Wiley
